build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/DownloadUri.cs

```diff
@@ -65,9 +74,12 @@ public override bool Execute ()
 
 			Log.LogMessage (MessageImportance.Normal, $"Downloading `{uri}` to `{tempPath}`.");
 			try {
-				using (var s = await client.GetStreamAsync (uri))
-				using (var o = File.OpenWrite (tempPath)) {
-					await s.CopyToAsync (o);
+				using (var r = await client.GetAsync (uri, source.Token)) {
+					r.EnsureSuccessStatusCode ();
+					using (var s = await r.Content.ReadAsStreamAsync ())
+					using (var o = File.OpenWrite (tempPath)) {
+						await s.CopyToAsync (o, 4096, source.Token);
```

Comment:
Should we also insert an `r.EnsureSuccessStatusCode()` before this statement as well?  I don't see why we'd need to wait for `CopyToAsync()` to *finish* before aborting the entire process...

Reply:
`CopyToAsync` is getting the `CancellationToken` here, so it should quit before writing the entire file.

When I was testing, Ctrl+C seemed instantaneous.